File: mobile/app/src/main/java/com/praeter/data/local/model/Ancient.java
package com.praeter.data.local.model;

import android.location.Location;

import org.parceler.Parcel;
import org.parceler.ParcelConstructor;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Setter
@Getter
@ToString
@Parcel(Parcel.Serialization.BEAN)
public class Ancient {

    String name;

    double latitude;
    double longitude;

    Location location;

    public Ancient() {
    }

    public Ancient(String name) {
        this.name = name;
    }

    public Ancient(String name, double latitude, double longitude) {
        this.name = name;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public Ancient(String name, Location location) {
        this.name = name;
        this.location = location;
        this.latitude = location.getLatitude();
        this.longitude = location.getLongitude();
    }
}

File: mobile/app/src/main/java/com/praeter/data/local/model/Ancient.kt
package com.praeter.data.local.model

import android.location.Location
import android.os.Parcelable
import com.squareup.moshi.JsonClass
import kotlinx.parcelize.Parcelize


@JsonClass(generateAdapter = true)
@Parcelize
class Ancient(
    var name: String? = null,
    var latitude: Double = 0.0,
    var longitude: Double = 0.0,
    var location: Location? = null
) : Parcelable {


    constructor(name: String) : this() {
        this.name = name
    }

    constructor(name: String, latitude: Double, longitude: Double) : this() {
        this.name = name
        this.latitude = latitude
        this.longitude = longitude
    }

    constructor(name: String, location: Location) : this() {
        this.name = name
        this.location = location
        latitude = location.latitude
        longitude = location.longitude
    }
}